### ISSUE TITLE
Cannot find info about player neutral. Throwing... Disaster happened. Reason: Cannot find info about player

### ISSUE THREAD

**Issue by nerozero** [vcmi/vcmi#5102]
**Describe the bug**
Game crash on AI move

**Game logs**
[VCMI_Client_log.txt](https://github.com/user-attachments/files/18151930/VCMI_Client_log.txt)

**To Reproduce**
Finish current step ( game save and map is in attachment)
[Daedlia-map-and-save.zip](https://github.com/user-attachments/files/18151968/Daedlia-map-and-save.zip)

**Expected behavior**
Not to crash I guess

**Actual behavior**
The game as well as my self-control died within 64bit hexadecimal agony

**Did it work earlier?**
for this particular map and place Idk, but I guess so ...

**Version**
Compiled from git, update
- OS:  Linux
- Version: VCMI 1.6.0.543ce94132b8e3820e02bdf5dee6f676d8202170

**Additional context**
My guess the soul processing instance in Hell refuses to accept the Girds (AI Teal player) soul 
```
Hero Charna has 1630 MP left
Player 5 (purple) ends turn
Player 6 (teal) starting turn, day 33

virtual const TerrainTile* CGameInfoCallback::getTile(int3, bool) const: (142 37 0) is not visible!

virtual std::vector<const CGObjectInstance*> CGameInfoCallback::getVisitableObjs(int3, bool) const: (142 37 0) is not visible!
Player 5 (purple) ended turn

virtual const TerrainTile* CGameInfoCallback::getTile(int3, bool) const: (142 37 0) is not visible!

virtual std::vector<const CGObjectInstance*> CGameInfoCallback::getVisitableObjs(int3, bool) const: (142 37 0) is not visible!
Player 6 (teal) ends turn
Player 7 (pink) starting turn, day 33
Hero Ajit has 1500 MP left
Hero Lorelei has 1539 MP left
Hero Dace has 1630 MP left
Hero Orrin has 851 MP left
Player 7 (pink) ends turn
Cannot find info about player neutral. Throwing...
Disaster happened.
Reason: Cannot find info about player
```

**Comment by Povelitelh3** [comment]
With this turn teal lost, maybe because of this. But I have no crash.

**Comment by nerozero** [comment]
Update, no need to load from save, just go and kill Girds, that will end-up in a crash 100% of times (to me). 

Few days ago I have compiled the e6cd3630 commit, and there is no glitch there. 

@Povelitelh3 Thanks, unfortunately I'm a bit ill at the moment, will try to compile that after I get better.

**Comment by IvanSavenko** [comment]
Should be fixed in latest build (including 1.6.0). However saves made with broken build are also broken.

**Comment by nerozero** [comment]
Awesome! Thanks so much for fast reaction!
